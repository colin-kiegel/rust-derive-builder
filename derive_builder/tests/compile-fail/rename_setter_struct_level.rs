#[macro_use]
extern crate pretty_assertions;
#[macro_use]
extern crate derive_builder;

#[derive(Debug, PartialEq, Default, Builder, Clone)]
#[builder(setter(name = "foo"))]
//~^ ERROR Unexpected field `name`
struct Lorem {
    ipsum: &'static str,
    pub dolor: &'static str,
}

#[test]
fn renamed_setter() {
    let x = LoremBuilder::default()
        .ipsum("ipsum")
        .foo("dolor")
        .build()
        .unwrap();

    assert_eq!(x,
               Lorem {
                   ipsum: "ipsum",
                   dolor: "dolor",
               });
}
