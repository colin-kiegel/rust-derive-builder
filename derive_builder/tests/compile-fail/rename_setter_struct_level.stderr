error: Unknown field: `name`
 --> tests/compile-fail/rename_setter_struct_level.rs:7:18
  |
7 | #[builder(setter(name = "foo"))]
  |                  ^^^^

error[E0433]: failed to resolve: use of undeclared type `LoremBuilder`
  --> tests/compile-fail/rename_setter_struct_level.rs:14:13
   |
14 |     let x = LoremBuilder::default()
   |             ^^^^^^^^^^^^ use of undeclared type `LoremBuilder`
